core/src/gateway_createfromjson.c

```diff
@@ -129,49 +131,72 @@ GATEWAY_HANDLE Gateway_CreateFromJson(const char* file_path)
     return gw;
 }
 
+static void rollbackModules(size_t success_modules_entries_count, GATEWAY_HANDLE gw, VECTOR_HANDLE modules_added_successfully)
```

Comment:
Why pass in `success_modules_entries_count` separately if that information is already part of the `VECTOR_HANDLE`?